optuna_dashboard/static/components/GraphContour.tsx

```diff
@@ -0,0 +1,259 @@
+import * as plotly from "plotly.js-dist"
+import React, { ChangeEvent, FC, useEffect, useState } from "react"
+import {
+  Grid,
+  FormControl,
+  FormLabel,
+  InputLabel,
+  MenuItem,
+  Select,
+} from "@material-ui/core"
+import { createStyles, makeStyles, Theme } from "@material-ui/core/styles"
+
+const plotDomId = "graph-contour"
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    formControl: {
+      marginBottom: theme.spacing(2),
+      marginRight: theme.spacing(2),
+    },
+  })
+)
+
+export const GraphContour: FC<{
+  study: StudyDetail | null
+}> = ({ study = null }) => {
+  const trials: Trial[] = study !== null ? study.trials : []
+  const filteredTrials = trials.filter(
+    (t) => t.state === "Complete" || t.state === "Pruned"
+  )
+
+  let paramNames = new Set<string>(trials[0].params.map((p) => p.name))
+  filteredTrials.forEach((t) => {
+    paramNames = new Set<string>(
+      t.params.filter((p) => paramNames.has(p.name)).map((p) => p.name)
+    )
+  })
+
+  const paramnames = Array.from(paramNames)
```

Comment:
These variable names (`paramNames` and `paramnames` ) looks confusing to me. Could you refactor this?

Reply:
Yeah sure!